interface.go

```diff
@@ -1,16 +1,23 @@
 package covalent
 
 import (
+	"github.com/ElrondNetwork/covalent-indexer-go/hyperBlock"
 	"github.com/ElrondNetwork/covalent-indexer-go/schema"
 	"github.com/ElrondNetwork/elrond-go-core/data"
 	"github.com/ElrondNetwork/elrond-go-core/data/indexer"
 	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+	"github.com/elodina/go-avro"
 )
 
 type DataHandler interface {
 	ProcessData(args *indexer.ArgsSaveBlockData) (*schema.BlockResult, error)
 }
 
+// HyperBlockProcessor shall handle hyper block processing into avro schema blocks
+type HyperBlockProcessor interface {
+	Process(hyperBlock *hyperBlock.HyperBlock) (*schema.BlockResult, error)
+}
+
 type Driver interface {
```

Comment:
missing comment here

Reply:
This is the "old" code. Will not add comments, since most of it will be deleted/refactored